webhooks-extension/src/components/WebhookDisplayTable/WebhookDisplayTable.js

```diff
@@ -171,37 +172,50 @@ export class WebhookDisplayTable extends Component {
           <Redirect to={this.props.match.url + "/create"} />
         )
       } else {
+        const { namespace } = this.props;
         // There are webhooks so display table
         const headers = [
           {
-            key: 'name',
-            header: 'Name',
+            key: "name",
+            header: "Name"
           },
           {
-            key: 'repository',
-            header: 'Git Repository',
+            key: "repository",
+            header: "Git Repository"
           },
           {
-            key: 'pipeline',
-            header: 'Pipeline',
-          },
-          {
-            key: 'namespace',
-            header: 'Namespace',
+            key: "pipeline",
+            header: "Pipeline"
           }
         ];
-    
-        let initialRows = []
+
+        if (namespace === ALL_NAMESPACES) {
+          headers.push({
+            key: "namespace",
+            header: "Namespace"
+          });
+        }
+
+        let initialRows = [];
         // Populate the data for the rows array from the data from the webhooks get request made on page load
-        this.state.webhooks.map(function (webhook, keyIndex) {
-          initialRows[keyIndex] = {
-            id: webhook['name']+"|"+webhook['namespace'],
-            name: webhook['name'],
-            repository: webhook['gitrepositoryurl'],
-            pipeline: webhook['pipeline'],
-            namespace: webhook['namespace'],
+        this.state.webhooks.map(function(webhook) {
```

Comment:
could also deconstruct the `webhook` param here since you only ever need individual fields, avoiding all the `webhook[foo]`

```js
({ gitrepositoryurl, name, namespace, pipeline }) => {

}
```